text/0000-multi-sig-term.md

```diff
@@ -0,0 +1,340 @@
+parties regulation might affect delegation.
+
+The resulting multi-sig terms have a compact binary representation and can be
+visualized in a easy to understand hierarchical tree-like structure. The
+multi-signature terms can be seen as a script-like DSL which is interpreted in
+order to validate them, therefore we also use the name multi-sig _script_ in
+this document.
+
+# Decision
+[Decision]: #decision
+
+A multi-sig term describes which combination of signatures of a transaction is
+required to:
+
+ - spend a multi-sig locked transaction output
+ - allow the withdrawal of a multi-sig locked rewards account
+ - allow changing the status of a multi-sig delegation reference
+
+## Format of Multi-Sig
+
+The recursive structure of a multi-sig term is of the form:
+
+ - `RequireSignature kh` where `kh` is a hashed key
+ - `RequireAllOf [msigs]` where `[msigs]` is a list of multi-sig terms
+ - `RequireAnyOf [msisg]` where `[msigs]` is a list of multi-sig terms
+ - `RequireMOf m [msig]` where `m` is a natural and `[msigs]` is a list of
+   multi-sig terms
+
+This forms a tree-like structure where each node is of the form
+`RequireSignature kh`. Leafs of the form `RequireAllOf` or `ReuquireMOf 0` with
+an empty argument list of multi-sig terms are redundant in the sense that they
+evaluate to `True`. Leaf terms of the form `RequireAnyOf` or `RequireMOf m`
+with `m>0` with an empty list of multi-sig terms is also redundant and evaluates
+to `False`.
+
+### Examples
+
+Assume we have key hashes for different participants: `aliceKH`, `bobKH`,
+`carlKH` and `dariaKH`. Then the simplest multi-signature term requires
+only Alice's signatures and looks as follows:
+
+```haskell
+aliceOnly = RequireSignature aliceKH
+```
+
+A multi-signature term that requires Alice's or Bob's signature looks as
+follows:
+
+```haskell
+aliceOrBob = RequireAnyOf [RequireSignature aliceKH, RequireSignature bobKH]
+```
+
+A script requiring both signatures looks as follows:
+
+```haskell
+aliceAndBob = RequireAllOf [RequireSignature aliceKH, RequireSignature bobKH]
+```
+
+Finally for an 2 out of 3 scheme for Alice, Bob and Carl, the multi-signature
+term looks as follows:
+
+```haskell
+aliceBobCarl2oo3 = RequireMOf 2 [RequireSignature aliceKH, RequireSignature bobKH,
+RequireSignature carlHK]
+```
+
+The terms can be combined recursively, e.g., requiring a signature of Bob and
+Alice or of Carl and Daria is expressed as follows:
+
+```haskell
+aliceAndBobOrCarlAndDaria = RequireAnyOf
+    [ RequireAllOf [RequireSignature aliceKH, RequireSignature bobKH]
+    , RequireAllOf [RequireSignature carlKH, RequireSignature dariaKH]]
+```
+
+`RequireAnyOf [msigs]` and `RequireMOf 1 [msigs]` are logically equivalent, just
+as `RequireAllOf [msigs]` and `RequireMOf (length msisg) [msigs]`. From a point
+of readability and to a lesser part also for encoding size, it makes sense to
+have the special case terms for _any_ and _all_.
```

Comment:
And it'll be more directly useful with later extensions, e.g. time locks and hash locks.

Reply:
I added this